internal/testutil/resources.go

```diff
@@ -0,0 +1,222 @@
+// Copyright 2016 Google Inc. All rights reserved.
+// Use of this source code is governed by the Apache 2.0
+// license that can be found in the LICENSE file.
+
+package testutil
+
+import (
+	"fmt"
+	"strconv"
+	"sync/atomic"
+	"testing"
+	"time"
+
+	gocontext "golang.org/x/net/context"
+	"golang.org/x/oauth2/google"
+
+	"strings"
+
+	"cloud.google.com/go/bigquery"
+	"cloud.google.com/go/storage"
+	rawbq "google.golang.org/api/bigquery/v2"
+	"google.golang.org/api/iterator"
+)
+
+var (
+	idCounter uint64
+	expiry    = time.Hour
+)
+
+const appPrefix = "golang_samples"
+
+// NextDataset returns the next unique dataset ID generated for the given namespace.
+func NextDataset(namespace string) string {
+	atomic.AddUint64(&idCounter, 1)
+	return fmt.Sprintf("%s_%s_dataset%d_%d", appPrefix, namespace, idCounter, time.Now().Unix())
+}
+
+// NextTable returns the next unique table ID generated for the given namespace
+func NextTable(namespace string) string {
```

Comment:
It seems like these could all be one function.

Or at least they should call one more general, unexported function.